packages/inertia-react/src/App.js

```diff
@@ -53,8 +54,8 @@ export default function App({
 
   return createElement(
     PageContext.Provider,
-    { value: page.props },
-    renderChildren({ Component: page.component, key: page.key, props: page.props })
+    { value: current.page },
+    renderChildren({ Component: current.component, key: current.key, props: current.page.props }),
   )
 }
 
```

Comment:
@sebastiandedeyne Ping! Just wanted you to be aware that I made these changes to the React adapter. 👍

Reply:
LGTM! `swapComponent`'s argument list looks a lot cleaner than `updatePage`!

Reply:
Awesome! ❤️